content/projects/gsoc/2021/projects/git-credentials-binding.adoc

```diff
@@ -50,35 +50,90 @@ They will be able to use the git command line options of their choice, including
+These variable bindings are also accessible when using `JGit/JGit with Apache HTTP Client` Git-Tool
+implementations in both freestyle project and pipeline job.
 
 ==== Git SSH Private Key Binding
 
-The `gitSshPrivateKey` implementation uses `isAtLeastVersion` method provided by `CliGitAPIImpl` to discover the minimum CLI git version that will be used in the sh, bat, and powershell commands within the `withCredentials` block.
-If the `cliGitBaseVersion` is greater than 2.3, then the GIT_SSH_COMMAND environment variable should be set and should include arguments that provide the path to the private key.
-If the `cliGitBaseVersion` is less than 2.3, then the current ssh technique in the git plugin should be used instead (needs more details of the current technique).
-The SSH_ASKPASS environment variable should be set to point to a file that is accessible to the agent workspace.
-That SSH_ASKPASS script should echo the passphrase if a passphrase is defined on the private key, just as is done today within the git plugin.
-The credential binding should write the ssh private key to the agent file system in a workspace specific temporary location and should set environment variables to provide the location of the ssh private key.
-Alternately, passphrase protected private keys should be converted by the plugin to not use a passphrase and the private key without passphrase should be written to the workspace specific temporary location instead of writing the private key with passphrase.
+The `gitSshPrivateKey` implementation provides git authentication support over *SSH* protocol
+using private key and passphrase credentials of a user. The binding uses two git specific environment
+variables depending upon the minimum CLI-git version
+
+** `GIT_SSH_COMMAND` - If version is greater than 2.3, then the GIT_SSH_COMMAND environment variable provides ssh command including the
+necessary options which are: path to the private key and host key checking to authenticate and connect git server without using an executable script.
+
+** `SSH_ASKPASS` - If version is less than 2.3, an executable script is attached to the variable which provides
+ssh command including the necessary options which are: path to the private key and host key checking to authenticate and connect git server
+
+Support for private key formats
+
+** The following key formats are supported through Jenkins plugins
+
+* `BouncyCastle API Plugin` - Supports decryption of PKCS#8 PEM encoded and PEM private keys.
+* `SSHD Plugin` - Provides a transitive dependency i.e. Apache SSHD-Core v2.7.0, which supports decryption of
+OpenSSH/RFC4716 formatted private keys.
+
+Support for encryption algorithms
+
+* The following encryption algorithms are supported in various formats discussed above
+
+** RSA
+** DSA
+** ECDSA
+** ED25519
+
+===== Decryption of private keys
+
+The passphrase protected private key needs to be decrypted before they could be used to perform
```

Comment:
before it could be used (I think the subject is "The passphrase" which is singular not plural).